cfme/v2v/infrastructure_mapping.py

```diff
@@ -0,0 +1,401 @@
+                   }
+                },
+                'network':{
+                    'Cluster (Default)': {
+                        'mappings': [
+                            {
+                                'sources':['VM Network','VMkernel'],
+                                'target':['ovirtmgmt']
+                            },
+                            {
+                                'sources':['DPortGroup'],
+                                'target':['Storage VLAN 33']
+                            }
+                        ]
+                    }
+                }
+            }
+    """
+
+    title = Text(locator='.//h4[contains(@class,"modal-title")]')
+    general = View.nested(InfraMappingWizardGeneralView)
+    cluster = View.nested(InfraMappingWizardClustersView)
+    datastore = View.nested(InfraMappingWizardDatastoresView)
+    network = View.nested(InfraMappingWizardNetworksView)
+    result = View.nested(InfraMappingWizardResultsView)
+
+    def after_fill(self, was_change):
+        if was_change:
+            self.result.close.click()
+
+
+class AddInfrastructureMappingView(InfraMappingForm):
+    form = InfraMappingWizardView()
+    form_title_text = VersionPicker(
+        {
+            Version.lowest(): "Infrastructure Mapping Wizard",
+            "5.10": "Create Infrastructure Mapping",
+        }
+    )
+
+    @property
+    def is_displayed(self):
+        return self.form.title.text == self.form_title_text
+
+
+class EditInfrastructureMappingView(InfraMappingForm):
+    form = InfraMappingWizardView()
+    form_title_text = "Edit Infrastructure Mapping"
+    save = Button("Save")
+
+    @property
+    def is_displayed(self):
+        return self.form.title.text == self.form_title_text
+
+
+@attr.s
+class InfrastructureMapping(BaseEntity, Updateable):
+    """Class representing v2v infrastructure mappings"""
+
+    name = attr.ib()
+    description = attr.ib(default=None)
+    form_data = attr.ib(default=None)
+
+    def update(self, updates):
+        view = navigate_to(self, "Edit", wait_for_view=20)
+        view.form.fill(updates)
+
+
+@attr.s
+class InfrastructureMappingCollection(BaseCollection):
+    """Collection object for Migration mapping object"""
+
+    ENTITY = InfrastructureMapping
+
+    def create(self, form_data):
+        view = navigate_to(self, "Add")
+        view.form.fill(form_data)
+        return self.instantiate(
+            name=form_data["general"]["name"],
+            description=form_data["general"].get("description", ""),
+            form_data=form_data,
+        )
+
+    def find_mapping(self, mapping):
+        view = navigate_to(self, "All")
+        if not self.appliance.version < "5.10":  # means 5.10+ or upstream
+            view.items_on_page.item_select("15")
+            if mapping.name in view.infra_mapping_list.read():
+                return True
+            # TODO: Remove While loop. It is a DIRTY HACK for now, to be addressed in PR 8075
```

Comment:
**QUESTION:** Can you check if we can remove this hack in this PR? if not, update the TODO comment to say "To be addressed in later PR"